test/unittests/execute_control_test.cpp

```diff
@@ -609,3 +609,149 @@ TEST(execute_control, return_from_block_stack_cleanup)
     ASSERT_EQ(ret.size(), 1);
     EXPECT_EQ(ret[0], 1);
 }
+
+TEST(execute_control, if_smoke)
+{
+    /*
+    (func (param $x i32) (result i32)
+      (local $y i32)
+      get_local $x
+      (if
+        (then
+          i32.const 4
+          set_local $y
+        )
+      )
+      get_local $y
+    )
+    */
+    const auto bin = from_hex(
+        "0061736D0100000001060160017F017F030201000A11010F01017F20000440410421010B20010B0010046E616D"
+        "650209010002000178010179");
+
+    for (const auto param : {0u, 1u})
+    {
+        constexpr uint64_t expected_results[]{
+            0,  // no if branch.
+            4,  // if branch.
+        };
+
+        const auto [trap, ret] = execute(parse(bin), 0, {param});
```

Comment:
I see some existing tests are doing this, but I think better not to parse in the loop.

Reply:
Fixed.